recipes/pugixml/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class PugiXmlConan(ConanFile):
+    name = "pugixml"
+    version = "1.10"
+    description = "Light-weight, simple and fast XML parser for C++ with XPath support"
+    topics = ("xml-parser", "xpath", "xml", "dom")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://pugixml.org/"
+    license = "MIT"
+    generators = "cmake"
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "header_only": [True, False],
+        "wchar_mode": [True, False],
+        "no_exceptions": [True, False]
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'header_only': False,
+        'wchar_mode': False,
+        'no_exceptions': False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.header_only:
+            if self.settings.os != 'Windows':
+                del self.options.fPIC
+            del self.options.shared
+            self.settings.clear()
```

Comment:
```suggestion
```
Clearing `self.settings` is not correct because the cmake helper will not have any info about the compiler available.